src/components/task-step/step-mixin.cjsx

```diff
@@ -22,8 +25,17 @@ module.exports =
   render: ->
     footer = @renderFooterButtons?() or @renderGenericFooter()
     {pinned} = @props
-
-    <CardBody className='task-step' footer={footer} pinned={pinned}>
+    <CardBody className={@state.classNames.join(' ')} footer={footer} pinned={pinned}>
       {@renderBody()}
       {@renderGroup?()}
     </CardBody>
+
+  componentDidMount:  -> @setBannerClass()
+  componentDidUpdate: -> @setBannerClass()
+
+  # If the rendered content has a matching element that will be used as the banner,
+  # add a "with-ui-banner" css class.  This way the component can be styled to match
+  setBannerClass: ->
+    has_banner = _.contains(@state.classNames, "with-ui-banner")
+    if not has_banner and this.getDOMNode().querySelector(":first-child[data-label]")
```

Comment:
Yes, I'm not sure either, but don't see any really better way to identify the element.  I'm using the `data-has-label` attribute for the css selector though, so I'll update this to match it.

The unless won't work like that though.  We need to test if the state has not been set and the dom node is present.  If we flip it to be an unless that'd read as:   

`unless has_banner and not this.getDOMNode().querySelector(":first-child[data-label]")` 

 Which I think is even more confusing.  Unless and not don't really mix, so I've flipped the hasBanner to be bannerMissing and it reads better now